src/SFML/Graphics/RenderTexture.cpp

```diff
@@ -143,13 +143,11 @@ bool RenderTexture::generateMipmap()
 ////////////////////////////////////////////////////////////
 bool RenderTexture::setActive(bool active)
 {
-    bool result = m_impl && m_impl->activate(active);
-
     // Update RenderTarget tracking
-    if (result)
-        RenderTarget::setActive(active);
+    if (m_impl && m_impl->activate(active))
+        return RenderTarget::setActive(active);
 
-    return result;
+    return false;
```

Comment:
Nice refactoring!

Could it happen though that `m_impl->activate()` succeeds and `RenderTarget::setActive()` fails? In that case, would we end up in an inconsistent state? (It wouldn't be worse than previous behavior, just asking).

Reply:
Good point -- I guess it is technically possible, I am not sure exactly how the behavior here could be improved. Maybe @binary1248 is more familiar with this part of the code?

Reply:
Well on first glance, if `RenderTarget::setActive(active);` fails something was going to go wrong anyway. The question is: Have users gotten used to the current behaviour enough that this change will break their code? I must admit, when I wrote that code I must have had something in mind, otherwise I wouldn't have went through the extra effort of saving `result` to a temporary. I just can't remember what the reason was. In any case, I think if we want to err on the side of caution I don't think we should change the logic unless it is proven it is currently definitely broken (which I haven't heard of yet).

Reply:
@binary1248: my assumption is that you meant to write 

```cpp
if (result)
    result = RenderTarget::setActive(active);
```

in order to propagate the result of the inner `setActive` call to the caller -- it's the only thing that makes sense to me, because the inner `setActive` can fail as well. My refactoring does exactly that.

Not propagating the value of the inner `setActive` is weird, because to the caller it looks like `RenderTexture::setActive` succeeded while in reality it didn't.

Reply:
That is also my thought when looking at the code now. 😁

But like I said, the current implementation that ignores the result of `RenderTexture::setActive` hasn't led to any bug reports (which doesn't have to mean anything).

I you want, we can make the change and see whether the visibly more correct logic will lead to reports.

Reply:
@binary1248: I think that would be best. I will take responsibility for any bug report 😜